tasks/main.yml

```diff
@@ -6,59 +6,41 @@
       set_fact:
         rsyslog_enabled: "{{ logging_enabled | d(true) }}"
 
-    - name: Set rsyslog_elasticsearch
+    - name: Set rsyslog_output_elasticsearch for the omelasticsearch output
       set_fact:
-        rsyslog_elasticsearch: "{{ rsyslog_elasticsearch | d([]) }} + {{ [ item.0 | combine( {'logs_collections_name': item.1.name} )] }}"
-      with_subelements:
-        - "{{ logging_outputs }}"
-        - logs_collections
+        rsyslog_output_elasticsearch: "{{ rsyslog_output_elasticsearch | d([]) }} + {{ [ item ] }}"
+      loop: "{{ logging_outputs }}"
```

Comment:
```
set_fact:
  rsyslog_output_elasticsearch: "{{ logging_outputs | selectattr('name','defined') | selectattr('type','defined') | selectattr('type','==','elasticsearch') }}"
```

Reply:
Ok. In this case, we can get rid of loop by using selectattr calls.  It simplifies the code.  I'd like to learn if there could be more benefits like performance?

Reply:
>  I'd like to learn if there could be more benefits like performance?

I don't know.